Improve mobile layout

wcedmisten opened the pull request:
Closes https://github.com/gis-ops/valhalla-app/issues/15

Improves the layout to be a bit more mobile friendly.

Minor regression: the drawer takes up the entire vertical space, which obscures part of the map on desktop. I tried fixing this by adding `height: undefined` to the style, but this broke the ability to scroll when the content expands past the screen.

# Changes
* use drawer for "Directions/Isochrones" modal
* use drawer for the settings modal
* don't show the settings page by default
* adds close button for the Directions/Isochrones drawer, allowing user to see the whole map
* adds a button to open the Directions/Isochrones drawer from the map view

You can also view a demo of this (only on North America) at https://map-demo.wcedmisten.dev, if you want to check it out on a true mobile device.

# Desktop Layout
![image](https://user-images.githubusercontent.com/4565200/202870790-de22abb1-2af8-4c90-b822-b60824314204.png)
![image](https://user-images.githubusercontent.com/4565200/202870813-c8089e5f-87cb-42dc-87ea-13c981ba95b6.png)
![image](https://user-images.githubusercontent.com/4565200/202870821-4585cdfc-a419-438d-a0b7-8b83020a4136.png)

# Mobile layout
![image](https://user-images.githubusercontent.com/4565200/202870848-be637f21-4865-4849-a67f-cd0479f30e74.png)
![image](https://user-images.githubusercontent.com/4565200/202870863-3a26102e-7048-40de-bec9-925efe315ad0.png)
![image](https://user-images.githubusercontent.com/4565200/202870867-7f4f828c-01da-41d1-82f5-eaeba416229c.png)

Sorry the commit history is a bit ugly because I had to revert the changes from my fork.

nilsnolde commented:
Wow, that's great! Thanks so much @wcedmisten ! We didn't have much time lately to care about this repo. I think this is a great start! I tried it on an iPhone 11 Pro and it looks really good. On Desktop it could be better I guess, but I'd argue it was worse before, now at least it's hideable sidebars. 

There's one more smaller thing I'd do (nothing related to your PR):
![image](https://user-images.githubusercontent.com/25637358/202875622-4d449cd3-c05a-4222-9805-7aafab407764.png)

The button group on the upper right: I think it's a weird arrangement, the settings button should be more prominent. IMO the settings button should be as big as the profile buttons and the "reset all" & "add via" buttons could be in the row below all locations. Then there's a bit more space in that row. Also, the settings button could be highlighted somehow, as that's hiding the whole meat of Valhalla.

What do you think @wcedmisten ? No problem if you'd like to skip that, can always do it in a later PR. Honestly this is great. We're about to apply for a renewal of the funding for the servers, so it's a great timing:)

nilsnolde commented on src/actions/directionsActions.js:
```diff
@@ -81,10 +81,10 @@ const fetchValhallaDirections = (valhallaRequest) => (dispatch) => {
   axios
     .get(VALHALLA_OSM_URL + '/route', config)
     .then(({ data }) => {
+      console.log(data)
```
debug statement

wcedmisten commented:
Thanks for the feedback! I went ahead and implemented the changes you suggested. It looked a little weird just having the reset/add buttons alone in the next row, so I moved them to the bottom for Directions and next to the search bar for Isochrones.

![image](https://user-images.githubusercontent.com/4565200/202905895-45b4fc6c-e9b4-47e5-8b52-102d900dc259.png)
![image](https://user-images.githubusercontent.com/4565200/202905911-d04039a5-6195-4321-bada-f3afc505410f.png)

nilsnolde reviewed (approved):
Great, totally agree, good job with the re-arrangement! Can’t wait to see it on the official site, I’ll deploy later tonight. Thanks a ton!
